src/Mantasflowers.Services/Services/Payment/PaymentService.cs

```diff
@@ -0,0 +1,170 @@
+﻿using AutoMapper;
+using Mantasflowers.Contracts.Payment.Request;
+using Mantasflowers.Contracts.Payment.Response;
+using Mantasflowers.Services.Services.Coupon;
+using Mantasflowers.Services.Services.Order;
+using Microsoft.EntityFrameworkCore;
+using Stripe;
+using Stripe.Checkout;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.Services.Services.Payment
+{
+    public class PaymentService : IPaymentService
+    {
+        private readonly IOrderService _orderService;
+        private readonly ICouponService _couponService;
+        private readonly SessionService _stripeSessionService;
+        private readonly Stripe.CouponService _stripeCouponService;
+        private readonly PromotionCodeService _promotionCodeService;
+        private readonly IMapper _mapper;
+
+        public PaymentService(IOrderService orderService,
+            ICouponService couponService,
+            SessionService stripeSessionService,
+            Stripe.CouponService stripeCouponService,
+            PromotionCodeService promotionCodeService,
+            IMapper mapper)
+        {
+            _orderService = orderService;
+            _couponService = couponService;
+            _stripeSessionService = stripeSessionService;
+            _stripeCouponService = stripeCouponService;
+            _promotionCodeService = promotionCodeService;
+            _mapper = mapper;
+        }
+
+        public async Task<PostCreateCheckoutSessionResponse> CreateCheckoutSessionAsync(PostCreateCheckoutSessionRequest request)
+        {
+            var executionStrategy = _orderService.CreateExecutionStrategy();
+
+            /**
+             * As a unit of repetition
+             * Read more on: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/implement-resilient-entity-framework-core-sql-connections
+             */
+            var response = await executionStrategy.ExecuteAsync(() => CreateCheckoutSessionBodyAsync(request));
+
+            return response;
+        }
+
+        private async Task<PostCreateCheckoutSessionResponse> CreateCheckoutSessionBodyAsync(PostCreateCheckoutSessionRequest request)
+        {
+            using var transaction = await _orderService.BeginTransactionAsync();
+            var session = new Session();
+
+            try
+            {
+                var order = await _orderService.CreateOrderAsync(request.Order);
+                order = await _orderService.GetDetailedOrderAsync(order.Id); // Is there a better solution?
```

Comment:
I think the original order instance gets populated by EF. So you might not actually need to refetch it (just speculation though)

Reply:
Populates everything, but not the related products, which are later required/used.